libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -316,6 +316,11 @@ public override async Task ContinueConversationAsync(ClaimsIdentity claimsIdenti
         }
 
         public override async Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, ConversationReference reference, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            await ContinueConversationAsync(claimsIdentity, reference, audience, null, callback, cancellationToken).ConfigureAwait(false);
+        }
+
+        public async Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, ConversationReference reference, string audience, string locale, BotCallbackHandler callback, CancellationToken cancellationToken)
         {
```

Comment:
I don't think this is right way to do this.  I think ConversationReference should capture the locale.

Reply:
@tomlm I actually considered that and ran it by @Stevenic, who recommended this approach since `ConversationReference` is a protocol data structure. Happy to make any changes necessary if we can get some consensus on which direction to go.

Reply:
Reviewed with the development team. Needs review by @dandriscoll.

Reply:
I don't know, within this part of the stack, what the protocol implications of putting the locale here are. If we set locale within the conversationRefererence, does it end up on the wire somewhere?

(Setting locale within conversationReference for storage reasons is the right approach, assuming no externalities introduced by setting it here.)

Reply:
Oh, interesting point.  Yes ConversationReference can be serialized into an activity, Activity in X can have a conversation reference to another conversation Y.  ConversationReference is the information you need to be able to switch to Y if you need to.  The pattern on this is that we need the original locale of the conversation Y to continue that conversation, so it seems like it should be in there, even if it gets serialized to wire.
@dandriscoll does that seem right to you?

---
In reply to: [409127783](https://github.com/microsoft/botbuilder-dotnet/pull/3739#discussion_r409127783) [](ancestors = 409127783)

Reply:
OK, I talked with Dan and we agreed it is appropriate to modify ConversationReference.  
Changes would be:
* Modify ConversationReference in schema assembly to have locale
* Modify CreateConversationReference to capture the locale.
Then all people get the proper behavior without having to change their code.
Rest of the APIS then are unchanged, except to use conversationReference.locale in appropriate place.